torch_xla/experimental/distributed_checkpoint.py

```diff
@@ -340,13 +345,12 @@ def _create_write_items_for_xla_sharded_tensor(
 def _create_write_items_for_cpu_shards(
     fqn: str, cpu_shards: _CpuShards) -> List[WriteItem]:
   items = []
-  for xla_shard in cpu_shards.shards:
+  for shard_ind, xla_shard in enumerate(cpu_shards.shards):
```

Comment:
Is this the bug you are fixing?

Reply:
Yeah, what we had originally was incorrect. It looks like the CPU tests didn't actually hit this codepath, and we weren't running on TPU CI.